lambdas/service/app.py

```diff
@@ -546,204 +546,229 @@ def repository_search(entity_type: str, item_id: str):
+                'order',
+                schema.optional(schema.enum('asc', 'desc')),
+                description='Whether results should be returned in ascending or descending order.'
+            ),
+            params.query(
+                'search_after',
+                schema.optional(str),
+                description='''
+                        The value of the \'sort\' field for the hit after which all
+                        results should be returned. Mutually exclusive with
+                        `search_before`.
+                    '''),
+            params.query(
+                'search_after_uid',
+                schema.optional(str),
+                description='''
+                        The value of the ElasticSearch UID corresponding to the hit
+                        above, if `search_after` is set.
+                    '''),
+            params.query(
+                'search_before',
+                schema.optional(str),
+                description='''
+                        The value of the \'sort\' field for the hit after which all
+                        results should be returned. Mutually exclusive with
+                        `search_after`.
+                    '''),
+            params.query(
+                'search_before_uid',
+                schema.optional(str),
+                description='''
+                        The value of the ElasticSearch UID corresponding to the hit
+                        above, if `search_before` is set.
+                    ''')
+        ],
+        'summary': f'Search the \'{entity_type}s\' index for entities of interest.',
+        'responses': {
+            '200': {
+                'description': format_description(f'''
+                    Paginated list of {entity_type}s that meet the search
+                    criteria ("hits"). The structure of these hits is documented
+                    under the [corresponding endpoint for a specific entity]\
+(#operations-Repository-get_repository_{entity_type}s__{entity_type}_id_).
+
+                    The `pagination` sections describes the total number of hits
+                    and total number of pages, as well as user-supplied search
+                    parameters such as page size and sorting behavior. It also
+                    provides as links for navigating forwards and backwards
+                    between pages of results.
+
+                    The `termFacets` section tabulates the occurrence of unique
+                    values within nested fields of the `hits` section. Not every
+                    nested field is tabulated, but the set of tabulated fields
+                    is consistent between entity types.
+                '''),
+                'content': {
+                    'application/json': {
+                        'schema': page_schema
+                    }
+                }
+            }
+        }
+    }
+
+
+def repository_id_spec(entity_type: str):
```

Comment:
This method takes in a parameter in order to be used by other entities, but I notice that only `files` uses it. Is that intended? I don't believe it is, but if so maybe there is no need to pass in an entity.